docs/proposals/hpa/federatedhpa-v2.md

```diff
@@ -0,0 +1,479 @@
+As a user, I deploy my application with HPA in multiple clusters. Since the unlimited scaling up of the application may lead to unexpected cloud costs and resource congestion that may cause service disruptions for other applications, so I expect to set the maximum number of instances for multi-cluster applications.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go into as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+1. The workloads/pods in different member clusters selected by the same HPA CR/resource share the load of the application equally. For example, 10 pods of the application are spread into two member clusters with distribution `cluster1: 3 pods, cluster2: 7 pods`, so the 3 pods in cluster1 take 3/10 of total requests and 7 pods in cluster2 take 7/10 of total requests. Scenarios that don't meet the restriction are not considered in this proposal.
+2. If the system does not meet the first requirement, some clusters' services may become overloaded while others remain underloaded.      
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate them? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. The HPA used in a single cluster can be seamlessly migrated to multiple clusters' FederatedHPA.
+
+## Design Details
+### Architecture
+For different scenarios, the architecture may be different:  
+
+#### HPA with CPU/Memory metrics
+![K8s native metrics](./statics/federatedhpa-native-metrics.png)  
+The steps are described as follows:
+1. HPA controller query metrics via metrics API `metrics.k8s.io` with label selector periodically.
+2. `karmada-apiserver` get the metrics API query, and it will route to `karmada-metrics-adapter` via API service registration.
+3. `karmada-metrics-adapter` always list&watch the member clusters' pods, when the requests come, it could query the metrics from the target clusters(where the pod exists). After the metrics are collected, it will aggregate them and return it.
+4. HPA controller will calculate the desired replicas based on metrics and scale the workload directly.
+
+
+**HPA with custom metrics**:  
+![Custom metrics](./statics/federatedhpa-custom-metrics.png)  
+The steps are described as follows:
+1. HPA controller query metrics via metrics API `custom.metrics.k8s.io` with label selector periodically.
+2. `karmada-apiserver` get the metrics API query, and it will route to `karmada-metrics-adapter` via API service registration.
+3. `karmada-metrics-adapter` always list&watch the member clusters' pods, when the requests come, it could query the metrics from the target clusters(where the pod exists). After the metrics are collected, it will aggregate them and return it.
```

Comment:
Do you list&watch the `nodes` of the member cluster?

If there are pods with the same name in a member cluster, how should they be handled?

Reply:
> Do you list&watch the nodes of the member cluster?

Yes

> If there are pods with the same name in a member cluster, how should they be handled?

They are in different Lister cache, I think there are no conflicts.

Reply:
ok,

I think you should provide some explanations in the proposal.

Reply:
Got it

Reply:
Fixed, please take another look